### PR TITLE
Create form route and mockup

### PR DESCRIPTION
**Pull request by vitorschelb**
# Description

This PR Create form route and mockup form
- Create appDir route `/app/form`
- Create Form mockup
- Create form questions
- Create form TW classes
- Create onboardingUtils

## Next Steps

- Create generic button for form submits
- Add logic

### CODE REVIEW

File: src/app/app/form/page.tsx [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import OnboardingForm from "@/components/OnboardingForm";
+
+export default function Form() {
+  return (
+    <div className="min-h-full">
+      <div className="bg-white shadow">
+        <div className="mx-auto max-w-7xl px-4 py-6 sm:px-6 lg:px-8">
```
Comment (vitorschelb) [inline src/app/app/form/page.tsx:7]: TODO: Leave only `<OnBoardingForm />`

**Review by vitorschelb — COMMENTED** [review]
Temporary mockup for Forms created. 
Waiting for reviews to complete next steps!

File: src/app/app/form/layout.tsx [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+import { PropsWithChildren } from "react";
+import { getServerSession } from "next-auth";
+import { redirect } from "next/navigation";
+import { authOptions, makeRedirectURL, getOriginPath } from "@/shared/auth";
+
+export const metadata: Metadata = {
+  title: "PodCodar",
+};
+
+export default async function RootLayout({ children }: PropsWithChildren) {
```
Comment (marco-souza) [inline src/app/app/form/layout.tsx:11]: We can remove this file, as we already have this logic on `/src/app/app/layout.tsx`

File: src/app/app/form/page.tsx [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import OnboardingForm from "@/components/OnboardingForm";
```
Comment (marco-souza) [inline src/app/app/form/page.tsx:1]: I dont think is a good idea hava a URL like `/app/form`. Usually we have more than one form in an app, so I would prefer a route like `/app/onboarding`

File: src/components/OnboardingForm.tsx [hunk 4]
```diff
@@ -0,0 +1,163 @@
+"use client";
+export default function OnboardingForm() {
+  return (
+    <div className="absolute h-screen w-screen z-10 top-0 left-0 bg-slate-800 overflow-auto">
+      <form className="bg-[#5d417A] w-full md:w-9/12 mx-auto my-10 p-8 rounded-lg">
```
Comment (marco-souza) [inline src/components/OnboardingForm.tsx:5]: Wow, why use this color? `bg-[#5d417A]`

This was part of the form template?

File: src/components/OnboardingForm.tsx [hunk 5]
```diff
@@ -0,0 +1,163 @@
+"use client";
```
Comment (marco-souza) [inline src/components/OnboardingForm.tsx:1]: Why are you using `"use client";` here?

This component does not have states

File: src/components/OnboardingForm.tsx (hunk 4 above)
Comment (nicolasbrandao) [reply src/components/OnboardingForm.tsx:5]: We picked this color from [figma](https://www.figma.com/file/NRnn6iLgO0oMXgZm1fPPPH/webapp?type=design&t=ygye4KG6Nk3OplBf-0)

Maybe we should create CSS/Tailwind vars for these colors

File: src/components/OnboardingForm.tsx (hunk 5 above)
Comment (nicolasbrandao) [reply src/components/OnboardingForm.tsx:1]: No special reason. We discussed this when coding and even though it works as an SSR component, for now, in the future as we use hooks for state management of the form itself, we're  likely making it a client-side component. But we don't need it for now, indeed

File: src/components/OnboardingForm.tsx (hunk 4 above)
Comment (nicolasbrandao) [reply src/components/OnboardingForm.tsx:5]: I'm gonna commit this with my next PR.

Comment (vitorschelb) [reply src/components/OnboardingForm.tsx:5]: Agree! Nice suggestion Nicolas!

**Review by vitorschelb — COMMENTED** [review]
- Deleted layout.
- Renamed Route

**Review by marco-souza — APPROVED** [review]
Good job! 🚢

**Review by marco-souza — COMMENTED** [review]
What do you think we merge it as it was previously? This PR it's already 9 commits long, it's not a good place to be.

We can continue the other tasks into other PRs, no need to conitnue working on this one

**Pull request merged**